### PR TITLE
Release 10.6.0beta1

### PR DESCRIPTION
**Pull request by jnweiger**

### CODE REVIEW

**Review by micbar — APPROVED** [review]

**Comment by jnweiger** [comment]
Cannot build docker images for 10.6 -- seems https://cloud.drone.io/owncloud-docker/server is blocked

**Review by iHerrmann77 — APPROVED** [review]

**Comment by micbar** [comment]
@phil-davis @jnweiger Somehow we have a different behavior here https://drone.owncloud.com/owncloud-docker/server/945/7/11

Seems that the new notifications are interfering.

But i wonder:

- The same tests pass if I execute them on my machine
- The same tests pass running on owncloud/core master branch.

**Comment by phil-davis** [comment]
On my local system I happen to have the message "There were problems with the code integrity check. More information..." that is displayed when I login as admin. I get a report like:
```
Results
=======
- core
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\MissingSignatureException
		- Signature data not found.
- federatedfilesharing
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\MissingSignatureException
		- Signature data not found.
- files
	- EXCEPTION
		- OC\IntegrityCheck\Exceptions\MissingSignatureException
		- Signature data not found.
...
```
I guess this happens for an install that is just a `git clone` for development.

When I run the feature:
```
Feature: admin apps settings
  As an admin
  I want to be able to manage apps settings on the ownCloud server
  So that I can enable or disable apps on the ownCloud server

  Background:                                                           # /home/phil/git/owncloud/core/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:8
INFORMATION: timed out waiting for ajax calls to start
    Given the administrator has browsed to the admin apps settings page # WebUIAdminAppsSettingsContext::theAdminBrowsesToTheAdminAppsSettingsPage()
      │ INFORMATION: timed out waiting for ajax calls to start

  @smokeTest
  Scenario: admin disables an app                                  # /home/phil/git/owncloud/core/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:12
    Given app "comments" has been enabled                          # FeatureContext::appHasBeenDisabled()
    When the administrator disables app "comments" using the webUI # WebUIAdminAppsSettingsContext::theAdminDisablesAppUsingTheWebUI()
      element click intercepted: Element <input class="enable" type="submit" data-appid="comments" data-active="true" value="Disable"> is not clickable at point (619, 834). Other element would receive the click: <div class="row">...</div>
        (Session info: chrome=80.0.3987.149)
      Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'
      System info: host: 'a490f324a0f2', ip: '172.17.0.3', os.name: 'Linux', os.arch: 'amd64', os.version: '5.4.0-51-generic', java.version: '1.8.0_242'
      Driver info: driver.version: unknown (Exception)
    Then app "comments" should be disabled                         # FeatureContext::appShouldBeDisabled()
SCENARIO RESULT: (fail)

  @smokeTest
  Scenario: admin enables an app                                  # /home/phil/git/owncloud/core/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:18
INFORMATION: timed out waiting for ajax calls to start
    Given the administrator has browsed to the admin apps settings page # WebUIAdminAppsSettingsContext::theAdminBrowsesToTheAdminAppsSettingsPage()
      │ INFORMATION: timed out waiting for ajax calls to start
    Given app "comments" has been disabled                        # FeatureContext::appHasBeenDisabled()
    And the administrator has browsed to the disabled apps page   # WebUIAdminAppsSettingsContext::theAdminHasBrowsedToDisabledAppsPage()
    When the administrator enables app "comments" using the webUI # WebUIAdminAppsSettingsContext::theAdminEnablesAppUsingTheWebui()
      element click intercepted: Element <input class="enable" type="submit" data-appid="comments" data-active="false" value="Enable"> is not clickable at point (317, 834). Other element would receive the click: <div id="notification" style="display: inline-block;">...</div>
        (Session info: chrome=80.0.3987.149)
      Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'
      System info: host: 'a490f324a0f2', ip: '172.17.0.3', os.name: 'Linux', os.arch: 'amd64', os.version: '5.4.0-51-generic', java.version: '1.8.0_242'
      Driver info: driver.version: unknown (Exception)
    Then app "comments" should be enabled                         # FeatureContext::appShouldBeEnabled()
SCENARIO RESULT: (fail)

--- Failed scenarios:

    /home/phil/git/owncloud/core/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:12
    /home/phil/git/owncloud/core/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:18

2 scenarios (2 failed)
9 steps (5 passed, 2 failed, 2 skipped)
1m12.71s (28.35Mb)
```
The comments app enable/disable buttons are obscured by the notification message.

If the test runs with a bigger browser window, then the comments app will be nicely visible and the test will pass.

I will think how best to work around this...

I guess that the notification comes because 10.6.0beta1 is not signed (or has not been signed at this stage, or...)

**Comment by micbar** [comment]
@phil-davis 
@jnweiger is investigating the build process. Seems that we have problems with the build of the complete tarball.

**Comment by phil-davis** [comment]
Note: when testing unsigned git core or apps that are fetched directly from git, the code in `lib/private/IntegrityCheck/Checker.php` `isCodeCheckEnforced()` looks at `version.php`
```
// The ownCloud channel
$OC_Channel = 'git';
```
And if the channel is `git`  then it does not enforce the integrity check.

IMO that is why, in ordinary CI, that we do not get trouble in the UI tests of admin features - the integrity check notification does not appear.

**Comment by jnweiger** [comment]
Not much luck with whe new tarball: https://drone.owncloud.com/owncloud-docker/server/947/7/11 explodes with

```
@smokeTest
  Scenario: admin enables an app                                  # /drone/src/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:18
    Given app "comments" has been disabled                        # FeatureContext::appHasBeenDisabled()
    And the administrator has browsed to the disabled apps page   # WebUIAdminAppsSettingsContext::theAdminHasBrowsedToDisabledAppsPage()
    When the administrator enables app "comments" using the webUI # WebUIAdminAppsSettingsContext::theAdminEnablesAppUsingTheWebui()
      element click intercepted: Element <input class="enable" type="submit" data-appid="comments" data-active="false" value="Enable"> is not clickable at point (317, 834). Other element would receive the click: <div id="notification" style="display: inline-block;">...</div>
        (Session info: chrome=74.0.3729.169)
        (Driver info: chromedriver=74.0.3729.6 (255758eccf3d244491b8a1317aa76e1ce10d57e9-refs/branch-heads/3729@{#29}),platform=Linux 4.15.0-112-generic x86_64) (Exception)
    Then app "comments" should be enabled                         # FeatureContext::appShouldBeEnabled()
SCENARIO RESULT: (fail)

--- Failed scenarios:

    /drone/src/tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature:18

**Comment by micbar** [comment]
@phil-davis Yes. Also interested in these local steps. I can see the `signature.json` . On a first glance it looks sane.

**Comment by phil-davis** [comment]
> How do you run the tests locally?

Somehow I have a local install that tells me there is a code integrity check problem. Actually I don't know why it does that, because it is a git clone and integrity check should be ignored in that case.

If you can make your local install complain about "integrity check" then:
```
make test-acceptance-webui BEHAT_FEATURE=tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature
```

**Comment by jnweiger** [comment]
@micbar @phil-davis Most of the CI pipelines in oauth2 run on php-71 and fail. the 7.2 and 7.3 pipelines work fine. 
I suggest to drop 7.1 there and add 7.4 instead.

**Comment by jnweiger** [comment]
Trying to locally on ubuntu-18.04 and remote on a hetzner box with ubuntu-20.04:
```
git checkout v10.6.0beta1
make test-acceptance-webui BEHAT_FEATURE=tests/acceptance/features/webUIAdminSettings/adminAppsSettings.feature
```
Filling in the php dependencies as per https://doc.owncloud.com/server/10.5/admin_manual/installation/ubuntu_18_04.html

The make command eventually fails with:
```
...
./tests/acceptance/run.sh --type webUI
Script path: /home/testy/src/github/owncloud/core/tests/acceptance
Using php inbuilt server for running scenario ...
8080
25636
8180
25637
PHP 7.2.24-0ubuntu0.18.04.7 Development Server started at Mon Oct 26 19:55:52 2020
Listening on http://localhost:8080
Document root is /home/testy/src/github/owncloud/core
Press Ctrl-C to quit.
PHP 7.2.24-0ubuntu0.18.04.7 Development Server started at Mon Oct 26 19:55:52 2020
Listening on http://localhost:8180
Document root is /home/testy/src/github
Press Ctrl-C to quit.
[Mon Oct 26 19:55:57 2020] 127.0.0.1:37150 [302]: /ocs/v2.php/apps/testing/api/v1/occ
-:2: parser error : Start tag expected, '<' not found

^
[Mon Oct 26 19:55:57 2020] 127.0.0.1:37152 [302]: /ocs/v2.php/apps/testing/api/v1/occ
-:2: parser error : Start tag expected, '<' not found

^
Could not set mail_domain on http://localhost:8080/ocs/v2.php/apps/testing/api/v1/occ. Result:
''
[Mon Oct 26 19:55:57 2020] 127.0.0.1:37154 [302]: /ocs/v2.php/apps/testing/api/v1/occ
-:2: parser error : Start tag expected, '<' not found

^
runsh: Exit code of main run: 
Makefile:215: recipe for target 'test-acceptance-webui' failed
make: *** [test-acceptance-webui] Error 1
```

**Comment by phil-davis** [comment]
The acceptance tests need the testing app to be installed and enabled. That is likely the first problem. The acceptance test code sets up the system-under-test with various default settings, using the testing app to do that.

It should work letting it run like above - letting it start a PHP dev server "in-line"... but actually I don't think anyone really does that on a day-to-day basis. I start a PHP dev server in another terminal window. Other people have Apache running with their development install being served by Apache...

You will need a selenium-browser ready - https://doc.owncloud.com/server/developer_manual/testing/ui-testing.html should have information on getting UI tests running.

I expect that the tests will pass fine with a fresh local install from a git clone. The problem will be something in the tarball that causes some notification to be displayed for the unpacked-and-installed tarball.

**Comment by jnweiger** [comment]
Almost there: 
```
su-exec www-data php /var/www/owncloud/occ config:system:set integrity.check.disabled --value true --type bool
ownCloud is not installed - only a limited number of commands are available


There are no commands defined in the "config:system" namespace
```

Let's try the same with a custom config.php snippet...

**Comment by phil-davis** [comment]
Or you could add a step between `wait(config) + ui(config)` that uses `owncloudci/php` and just does the command - `occ`  commands should work find at that point.

**Comment by micbar** [comment]
Sorry guys 😬 Sometimes it is hard.

File: .drone.star [hunk 1]
```diff
@@ -275,7 +275,8 @@ def docker(config):
               'OWNCLOUD_DB_NAME': 'owncloud',
             },
             'commands': [
-               'echo "\\$CONFIG = array(\'integrity.check.disabled\' => true);" > /var/www/owncloud/config/integrity-check-disabled.config.php',
+               # FIXME: Unfortunately, this config.php snippet does not get picked up.
+               'echo "\$CONFIG = array(\'integrity.check.disabled\' => true);" > /var/www/owncloud/config/integrity-check-disabled.config.php',
```
Comment (iHerrmann77) [inline .drone.star:279]: the backslash before $CONFIG forgotten or deleted?

**Comment by phil-davis** [comment]
1 CI pipeline had a silly 10-minute timeout. I restarted drone and now it passes - so we have a known-good, if hacky, solution.

**Review by iHerrmann77 — APPROVED** [review]

**Pull request merged**